### ISSUE TITLE
Unable to add weapon mod to weapon

### ISSUE THREAD

**Issue by ConnorMartin** [CleverRaven/Cataclysm-DDA#41926]
# Describe the bug

Weapon mods such as the rifle scope fail to find valid weapons to attach to

# Steps To Reproduce

Spawn a SCAR-L and a rifle scope, with appropriate skills.
Activate the rifle scope.
The UI will state that "there are no available choices"

# Expected behavior

The rifle scope should be attached to the weapon.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.E-3726-g8aaf68f [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Alternative Map Key [alt_map_key]
]

**Comment by LaVeyanFiend** [comment]
Cannot replicate on latest experimental(B10815).
![image](https://user-images.githubusercontent.com/51099123/86806943-5fc24680-c047-11ea-9fd9-4855978eca4e.png)

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-3783-g754b89e [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

**Comment by hexagonrecursion** [comment]
Possibly related https://github.com/CleverRaven/Cataclysm-DDA/issues/40376#issuecomment-626115938
> Did you try with the gun unloaded? I believe this is a requirement. I was able to attach gunmods to a weapon in the latest build, but only after removing any cartidges.

I'll repeat here what I said in the comments under that issue
> The UI should be improved. It should display loaded guns in the menu, but grayed out with a message explaining that you should unload the gun first. Otherwise it is hard for a new player to figure this mechanic out.

**Comment by ConnorMartin** [comment]
Here's me trying this in 0.D. It looks like there was a UI prompt for trying to mod a loaded gun that was removed sometime in the last half year.
![image](https://user-images.githubusercontent.com/19824381/86809609-0a3b6900-c04a-11ea-9780-e28ffd30db98.png)
@hexagonrecursion yeah, the gun was loaded- thanks for the help.
Should I create a new issue for the lack of "cannot mod a loaded gun" prompt?

**Comment by hexagonrecursion** [comment]
> Should I create a new issue for the lack of "cannot mod a loaded gun" prompt?

@ConnorMartin That is what I would do

**Comment by KorGgenT** [comment]
LVF can't confirm, and even if this is still a bug i'm confident that #41917 fixes it. if it doesn't, after it's merged a new issue can be opened.
